File: 1000/1430C/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    while (t--) 
    {
        int n;
        cin >> n;

        // answer will always be 2
        cout << 2 << "\n";  
        int a = n;
        int b = n - 1;

        // Generate pairs 
        // a = n, b = n - 1
        for (int i = 1; i < n; i++) 
        {
            cout << a << " " << b << "\n";
            a = (int)ceil((a + b) / 2.0);
            b = n - 1 - i;
        }
    }

    return 0;
}

File: 1000/1430C/main.java
import java.util.*;
public class main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();

        while(t-- > 0)
        {
            int n = sc.nextInt();

            // answer will always be 2
            System.out.println(2);
            int a = n;
            int b = n-1;

            // Generate pairs
            for(int i=1;i<n;i++)
            {
                System.out.println(a + " " + b);
                a = (int)Math.ceil((a+b)/2.0);
                b = n-1-i;
            }
        }
    }
}